django/db/migrations/autodetector.py

```diff
@@ -1080,11 +1080,31 @@ def generate_removed_fields(self):
             self._generate_removed_field(app_label, model_name, field_name)
 
     def _generate_removed_field(self, app_label, model_name, field_name):
+        field = self.from_state.models[app_label, model_name].get_field(field_name)
+        # You can't just remove NOT NULL fields with no default or fields
+        # which don't allow empty strings as default because the reverse
+        # migration will need a default value to populate existing records in
+        # the database while adding the field back.
+        temp_default = models.NOT_PROVIDED
+        time_fields = (models.DateField, models.DateTimeField, models.TimeField)
+        if (
+            not field.null
+            and not field.has_default()
+            and not field.many_to_many
+            and not (field.blank and field.empty_strings_allowed)
+            and not (
+                isinstance(field, time_fields)
```

Comment:
Checking the type is unnecessary. Often duck typing is used in Django; presence of auto_now or auto_now_add is enough.

Reply:
But fields other than time_fields don't have auto_now attribute. won't they throw an error? It is used even in field addition, see [here.](https://github.com/DevilsAutumn/django/blob/main/django/db/migrations/autodetector.py#L1046)

Reply:
Maybe let's use:
```python
getattr(field, "auto_now", False) or getattr(field, "auto_now_add", False)
```
This is the same as `BaseDatabaseSchemaEditor._effective_default`.
I think we have everything from `BaseDatabaseSchemaEditor._effective_default`, I tested `GeneratedField` case but these are covered by `not field.null`.